### PR TITLE
Fingerprint authentication added in the passcode page 

### LINKED ISSUE #1208: No fingerprint authentication available, only passcode authentication
**Summary:**

There is no fingerprint authentication but only passcode.

**Expected behaviour:**

When the app asks for passcode, the user should also be able to authenticate using his fingerprint

**Observed behaviour:**

The PassCode feature only asks for passcode user saved.

**Device and Android version:**

OnePlus 6
Android 10
stock ROM

### PR DESCRIPTION
**Pull request by PratyushSingh07**
Fixes #1208

https://user-images.githubusercontent.com/90026952/216406124-2182f61f-ff11-4a1f-8c00-9cb7c345c0f4.mp4
Above video can be summarized by the below points : 
- Fingerprint authentication is opened during the passcode activity
- Authentication reappears after exiting the app
- when setting up the new pin it doesn't ask for the fingerprint scanner

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
@IOhacker @rahul-gill pardon my naivete and the constant tagging but can you please review this PR as well

File: app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.kt [hunk 1]
```diff
@@ -15,15 +30,42 @@ import org.mifos.mobile.utils.Constants
 import org.mifos.mobile.utils.MaterialDialog
 import org.mifos.mobile.utils.Toaster
 
+@RequiresApi(Build.VERSION_CODES.M)
```
Comment (rahul-gill) [inline app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.kt]: Will cause crash on api lower than M. Pull out the biometric thing seperate like [this](https://gist.github.com/rahul-gill/cb2c5f2fe564b07bb3f982d42dc05ea5), check if device has biometric capabilities and then based on that, show the biometric prompt

Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/activities/PassCodeActivity.kt]: @rahul-gill on it

**Comment by PratyushSingh07** [comment]
@rahul-gill kindly review the changes in the PR

**Comment by PratyushSingh07** [comment]
Closing this PR in accordance with the latest PR #2034

**Pull request closed without merging**